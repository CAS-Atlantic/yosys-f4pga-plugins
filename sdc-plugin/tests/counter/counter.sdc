create_clock -name clk -period 10.0 clk
