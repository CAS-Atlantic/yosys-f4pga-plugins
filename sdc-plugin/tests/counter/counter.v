module top(input clk,
	input [1:0] in,
	output [1:0] out);

reg [1:0] cnt = 0;

always @(posedge clk) begin
	cnt <= cnt + 1;
end

assign out = {cnt[0], in[0]};
endmodule
/*
module dut();
reg clk;
wire [1:0] out;

top dut(.clk(clk), .in(2'b11), .out(out));
initial begin
	$dumpfile("test.vcd");
	$dumpvars(0,dut);
	clk = 0;
end

always
begin
	clk = #5 !clk;
end
endmodule
*/
