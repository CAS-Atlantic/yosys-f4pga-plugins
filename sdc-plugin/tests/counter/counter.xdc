set_property LOC E3 [get_ports clk]
set_property IOSTANDARD LVCMOS33 [get_ports clk]

set_property LOC J13 [get_ports {in[0]}]
set_property IOSTANDARD LVCMOS33 [get_ports {in[0]}]

set_property LOC J14 [get_ports {in[1]}]
set_property IOSTANDARD LVCMOS33 [get_ports {in[1]}]

set_property LOC K15 [get_ports {out[0]}]
set_property IOSTANDARD LVCMOS33 [get_ports {out[0]}]

set_property LOC K16 [get_ports {out[1]}]
set_property IOSTANDARD LVCMOS33 [get_ports {out[1]}]

