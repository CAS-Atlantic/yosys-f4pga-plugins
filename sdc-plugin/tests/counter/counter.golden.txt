clk_int_1 clk ibuf_proxy_out {$auto$clkbufmap.cc:247:execute$1918} {$auto$clkbufmap.cc:247:execute$1920}
